Mathlib/CategoryTheory/Monad/EquivMon.lean

```diff
@@ -60,99 +56,72 @@ variable (C)
+  μ' := M.mul
+  left_unit' := fun X => by
+    -- Porting note: now using `erw`
+    erw [← NatTrans.id_hcomp_app M.one, ← NatTrans.comp_app, M.mul_one]
+    rfl
+  right_unit' := fun X => by
+    -- Porting note: now using `erw`
+    erw [← NatTrans.hcomp_id_app M.one, ← NatTrans.comp_app, M.one_mul]
+    rfl
+  assoc' := fun X => by
+    rw [← NatTrans.hcomp_id_app, ← NatTrans.comp_app]
+    -- Porting note: had to add this step:
+    erw [M.mul_assoc]
+    simp
 #align category_theory.Monad.of_Mon CategoryTheory.Monad.ofMon
 
+-- Porting note: `@[simps]` fails to generate `ofMon_obj`:
+@[simp] lemma ofMon_obj (M : Mon_ (C ⥤ C)) (X : C) : (ofMon M).obj X = M.X.obj X := rfl
+
 variable (C)
 
 /-- Passing from `Mon_ (C ⥤ C)` to `Monad C` is functorial. -/
 @[simps]
 def monToMonad : Mon_ (C ⥤ C) ⥤ Monad C where
   obj := ofMon
-  map _ _ f :=
-    {-- `finish` closes this goal
-        f.Hom with
-      app_η' := by
+  map {X Y} f :=
+    { f.hom with
+      app_η := by
         intro X
-        erw [← nat_trans.comp_app, f.one_hom]
+        erw [← NatTrans.comp_app, f.one_hom]
         rfl
-      app_μ' := by
-        intro X
-        erw [← nat_trans.comp_app, f.mul_hom]
-        simpa only [nat_trans.naturality, nat_trans.hcomp_app, assoc, nat_trans.comp_app,
-          of_Mon_μ] }
+      app_μ := by
+        intro Z
+        erw [← NatTrans.comp_app, f.mul_hom]
+        dsimp
+        simp only [NatTrans.naturality, NatTrans.hcomp_app, assoc, NatTrans.comp_app,
+          ofMon_μ] }
 #align category_theory.Monad.Mon_to_Monad CategoryTheory.Monad.monToMonad
 
-namespace MonadMonEquiv
-
-variable {C}
-
-/-- Isomorphism of functors used in `Monad_Mon_equiv` -/
-@[simps (config := { rhsMd := semireducible })]
-def counitIso : monToMonad C ⋙ monadToMon C ≅ 𝟭 _ where
-  Hom := { app := fun _ => { Hom := 𝟙 _ } }
-  inv := { app := fun _ => { Hom := 𝟙 _ } }
-  hom_inv_id' := by ext; simp
-  -- `obviously` provides these, but slowly
-  inv_hom_id' := by ext; simp
-#align category_theory.Monad.Monad_Mon_equiv.counit_iso CategoryTheory.Monad.MonadMonEquiv.counitIso
-
-/-- Auxiliary definition for `Monad_Mon_equiv` -/
-@[simps]
-def unitIsoHom : 𝟭 _ ⟶ monadToMon C ⋙ monToMonad C where app _ := { app := fun _ => 𝟙 _ }
-#align category_theory.Monad.Monad_Mon_equiv.unit_iso_hom CategoryTheory.Monad.MonadMonEquiv.unitIsoHom
-
-/-- Auxiliary definition for `Monad_Mon_equiv` -/
-@[simps]
-def unitIsoInv : monadToMon C ⋙ monToMonad C ⟶ 𝟭 _ where app _ := { app := fun _ => 𝟙 _ }
-#align category_theory.Monad.Monad_Mon_equiv.unit_iso_inv CategoryTheory.Monad.MonadMonEquiv.unitIsoInv
-
-/-- Isomorphism of functors used in `Monad_Mon_equiv` -/
-@[simps]
-def unitIso : 𝟭 _ ≅ monadToMon C ⋙ monToMonad C where
-  Hom := unitIsoHom
-  inv := unitIsoInv
-  hom_inv_id' := by ext; simp
-  -- `obviously` provides these, but slowly
-  inv_hom_id' := by ext; simp
-#align category_theory.Monad.Monad_Mon_equiv.unit_iso CategoryTheory.Monad.MonadMonEquiv.unitIso
-
```

Comment:
Where did these go? Did they move somewhere?

Reply:
They're inlined into the definition of  `monadMonEquiv`, since the automation in Lean4 seems to fill in the proof obligations better. They're just the fields of `MonadMonEquiv`, so I'm happy to get rid of them. Maybe it should be backported, but this file's a leaf so I say LGTM

Reply:
Ok, lgtm